primer3/src/libprimer3/thalflex.c

```diff
@@ -0,0 +1,3393 @@
+/* libprimer3/thalflex.c
+
+Source original thal.c
+
+ Copyright (c) 1996,1997,1998,1999,2000,2001,2004,2006,2007,2009,2010,
+               2011,2012,2016
+ Whitehead Institute for Biomedical Research, Steve Rozen
+ (http://purl.com/STEVEROZEN/), and Helen Skaletsky
+ All rights reserved.
+
+       This file is part of primer3 software suite.
+
+       This software suite is is free software;
+       you can redistribute it and/or modify it under the terms
+       of the GNU General Public License as published by the Free
+       Software Foundation; either version 2 of the License, or (at
+       your option) any later version.
+
+       This software is distributed in the hope that it will be useful,
+       but WITHOUT ANY WARRANTY; without even the implied warranty of
+       MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+       GNU General Public License for more details.
+
+       You should have received a copy of the GNU General Public License
+       along with this software (file gpl-2.0.txt in the source
+       distribution); if not, write to the Free Software
+       Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ OWNERS OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON A THEORY
+ OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+Modifications to thal.c to create thalflex.c
+Copyright (c) 2023. Ben Pruitt & Nick Conway;
+
+To enable non-global mutable variable data,  thal.c has been modified and split
+into 3 files, thalflex.c, thalflex.h, and thalflexsignatures.h to enhance code
```

Comment:
I'd vote that you add a little more color to this explanation and that it should live in one of the headers (if only in one location) or just be duplicated across `thalflex.c`, `thalflex.h`, `thalflexsignatures.h`

Reply:
I'll add more color to the `thalflex.c` header, but I prefer leaving the "what this file is" description in each

Reply:
done